Source/Plugins/NetworkEvents/NetworkEvents.cpp

```diff
@@ -535,89 +247,100 @@ void NetworkEvents::process (AudioSampleBuffer& buffer)
 {
     setTimestampAndSamples(CoreServices::getGlobalTimestamp(),0);
 
-    lock.enter();
+    ScopedLock lock(queueLock);
     while (! networkMessagesQueue.empty())
     {
-        StringTS msg = networkMessagesQueue.front();
+        const StringTS& msg = networkMessagesQueue.front();
         postTimestamppedStringToMidiBuffer (msg);
-        CoreServices::sendStatusMessage ( ("Network event received: " + msg.getString()).toRawUTF8());
         networkMessagesQueue.pop();
     }
-
-    lock.exit();
-}
-
-
-void NetworkEvents::opensocket()
-{
-    startThread();
 }
 
 
 void NetworkEvents::run()
 {
 #ifdef ZEROMQ
-    responder = zmq_socket (zmqcontext, ZMQ_REP);
-    String url= String ("tcp://*:") + String (urlport);
-    int rc = zmq_bind (responder, url.toRawUTF8());
-
-    if (rc != 0)
-    {
-        // failed to open socket?
-        std::cout << "Failed to open socket: " << zmq_strerror (zmq_errno()) << std::endl;
-        return;
-    }
-
-    threadRunning = true;
-    unsigned char* buffer = new unsigned char[MAX_MESSAGE_LENGTH];
-    int result = -1;
+    ScopedPointer<Responder> responder;
+    char buffer[MAX_MESSAGE_LENGTH];
```

Comment:
This should probably be a `HeapBlock` rather than an array on the stack, actually. I was thinking I could use a plain array since the size never changes, but I wasn't considering the general rule that large arrays should be on the heap.